kernel/src/transaction/mod.rs

```diff
@@ -530,18 +545,48 @@ pub struct PostCommitStats {
 #[derive(Debug)]
 pub enum CommitResult {
     /// The transaction was successfully committed.
-    Committed {
-        /// the version of the table that was just committed
-        version: Version,
-        /// The [`PostCommitStats`] for this transaction
-        post_commit_stats: PostCommitStats,
-    },
+    CommittedTransaction(CommittedTransaction),
     /// This transaction conflicted with an existing version (at the version given). The transaction
     /// is returned so the caller can resolve the conflict (along with the version which
     /// conflicted).
     // TODO(zach): in order to make the returning of a transaction useful, we need to add APIs to
     // update the transaction to a new version etc.
-    Conflict(Transaction, Version),
+    ConflictedTransaction(ConflictedTransaction),
+}
+
+#[derive(Debug)]
+pub struct CommittedTransaction {
+    transaction: Transaction,
+    /// the version of the table that was just committed
+    commit_version: Version,
+    /// The [`PostCommitStats`] for this transaction
+    post_commit_stats: PostCommitStats,
+}
+
+impl CommittedTransaction {
+    /// The version of the table that was just committed
+    pub fn version(&self) -> Version {
+        self.commit_version
+    }
+
+    /// The [`PostCommitStats`] for this transaction
+    pub fn post_commit_stats(&self) -> &PostCommitStats {
+        &self.post_commit_stats
+    }
+
+    /// Compute a new snapshot for the table at the commit version. Note this is generally more
+    /// efficient than creating a new snapshot from scratch.
+    pub fn post_commit_snapshot(&self, engine: &dyn Engine) -> DeltaResult<SnapshotRef> {
```

Comment:
If we know what we just commited, can we construct the snapshot without having to do any work at all? 

Not for this PR, but should we have a follow-up?

Reply:
actually realized this was more complex. removed and tracking in https://github.com/delta-io/delta-kernel-rs/issues/916